Source/Scene/Gltf3DTileContent.js

```diff
@@ -98,97 +126,406 @@ Object.defineProperties(Gltf3DTileContent.prototype, {
+    throw new DeveloperError(
+      "batchId is required and between zero and featuresLength - 1 (" +
+        (featuresLength - 1) +
+        ")."
+    );
+  }
+  //>>includeEnd('debug');
+
+  createFeatures(this);
+  return this._features[batchId];
 };
 
 Gltf3DTileContent.prototype.applyDebugSettings = function (enabled, color) {
-  this._model.color = color;
+  color = enabled ? color : Color.WHITE;
+  if (this.featuresLength === 0) {
+    this._model.color = color;
+  } else {
+    this._batchTable.setAllColor(color);
+  }
 };
 
 Gltf3DTileContent.prototype.applyStyle = function (style) {
-  var hasColorStyle = defined(style) && defined(style.color);
-  var hasShowStyle = defined(style) && defined(style.show);
-  this._model.color = hasColorStyle
-    ? style.color.evaluateColor(undefined, this._model.color)
-    : Color.clone(Color.WHITE, this._model.color);
-  this._model.show = hasShowStyle ? style.show.evaluate(undefined) : true;
+  if (this.featuresLength === 0) {
+    var hasColorStyle = defined(style) && defined(style.color);
+    var hasShowStyle = defined(style) && defined(style.show);
+    this._model.color = hasColorStyle
+      ? style.color.evaluateColor(undefined, this._model.color)
+      : Color.clone(Color.WHITE, this._model.color);
+    this._model.show = hasShowStyle ? style.show.evaluate(undefined) : true;
+  } else {
+    this._batchTable.applyStyle(style);
+  }
 };
 
 Gltf3DTileContent.prototype.update = function (tileset, frameState) {
+  var commandStart = frameState.commandList.length;
+
   // In the PROCESSING state we may be calling update() to move forward
   // the content's resource loading.  In the READY state, it will
   // actually generate commands.
-  var model = this._model;
-  var tile = this._tile;
-
-  model.modelMatrix = tile.computedTransform;
-  model.shadows = tileset.shadows;
-  model.imageBasedLightingFactor = tileset.imageBasedLightingFactor;
-  model.lightColor = tileset.lightColor;
-  model.luminanceAtZenith = tileset.luminanceAtZenith;
-  model.sphericalHarmonicCoefficients = tileset.sphericalHarmonicCoefficients;
-  model.specularEnvironmentMaps = tileset.specularEnvironmentMaps;
-  model.backFaceCulling = tileset.backFaceCulling;
-  model.debugWireframe = tileset.debugWireframe;
+  this._batchTable.update(tileset, frameState);
+
+  this._model.modelMatrix = this._tile.computedTransform;
+  this._model.shadows = this._tileset.shadows;
+  this._model.imageBasedLightingFactor = this._tileset.imageBasedLightingFactor;
+  this._model.lightColor = this._tileset.lightColor;
+  this._model.luminanceAtZenith = this._tileset.luminanceAtZenith;
+  this._model.sphericalHarmonicCoefficients = this._tileset.sphericalHarmonicCoefficients;
+  this._model.specularEnvironmentMaps = this._tileset.specularEnvironmentMaps;
+  this._model.backFaceCulling = this._tileset.backFaceCulling;
+  this._model.debugWireframe = this._tileset.debugWireframe;
```

Comment:
was there a reason for changing everything here to use underscore properties? I find it reads better to use the shorter names

Reply:
I prefer the old code as well. The reason for the change was to copy what `Batched3DModel3DTileContent` was doing. I'll change it back.